### PR TITLE
 AICA Cleanups & Internalize CPU Core

### PR DESCRIPTION
**Pull request by cam900**
aica.cpp : Minor cleanups, Internalize ARM7DI core, Add device_clock_changed, Add notes, Sample/Timer rate related input clock, Implement some registers, Fix suffixes for hexadecimal values, Add imperfect_features Because emulation aren't completed, Fix suffixes for hexadecimal values
aicadsp.cpp : Fix suffixes for hexadecimal values
dccons.cpp : Add notes, Add stereo(Both chip and output supports stereo), Minor cleanups, Add AICA clock
dc.cpp : Minor cleanups
hikaru.cpp : Add docs for aica ram size/sound board, Add AICA clock
naomi.cpp : Add docs for regional title difference, Fix suffixes for hexadecimal values, Add AICA clock

### CODE REVIEW

**Comment by p1pkin** [comment]
> simpler ADPCM decoding

please revert decoder changes. current implementation verified as (most closest to) real hardware.
yes, it must be exact in this way: (previous quant * new quant)/8, then clip, and only then apply sign, + previous output signal.
there is number of oddball ADPCM samples which requires this, for example Shenmue game, in either case output stream will be broken.

File: src/devices/sound/aica.cpp [hunk 1]
```diff
@@ -8,12 +8,14 @@
...
     - A third sample format (ADPCM) has been added
     - Some minor other tweeks (no EGHOLD, slighly more capable DSP)
+	- Internal ARM7DI Core
 */
 
 #include "emu.h"
 #include "aica.h"
 
 #define ICLIP16(x) (x<-32768)?-32768:((x>32767)?32767:x)
+#define CLIP(x,min,max) (x<min)?min:((x>max)?max:x)
```
Comment (cuavas) [inline src/devices/sound/aica.cpp]: This isn't safe with macro substitution when the arguments are expressions.  The macro above it isn't safe either, but there's no excuse for adding more unsafe macros.

**Comment by rb6502** [comment]
This contained over a dozen fatal compile errors so I reverted it.  It is of absolute importance that submitted change requests actually compile when applied to the tree.

**Comment by happppp** [comment]
@cam900 please try this PR again, but make sure to test it.

**Pull request merged**